larray_editor/editor.py

```diff
@@ -503,11 +503,12 @@ def update_title(self):
         self.setWindowTitle(' - '.join(title))
 
     def set_current_array(self, array, name):
-        if array is not self.current_array:
-            self.current_array = array
-            self.arraywidget.set_data(array)
-            self.current_array_name = name
-            self.update_title()
+        # we should NOT check that "array is not self.current_array" because this method is also called to
+        # refresh the widget value because of an inplace setitem
```

Comment:
OK but `set_current_array` is also called by `view_expr` which passes '<expr>' has name argument and overwrite the real name of the array.

Reply:
view_expr uses '<expr>' and this is perfectly fine since it is no longer the same array